### ISSUE TITLE
Logout notice should be more visible, instead of a mere notification (+ I was "logged out" for no reason?)

### ISSUE THREAD

**Issue by p0358** [ppy/osu#27407]
### Type

Game behaviour

#### 1. Non-noticeable logout
If you suddenly get logged out, all you get is a simple notification (see below). The problem is: people don't immediately notice/look into notifications, especially that they tend to contain spam such as notices that beatmap was downloaded, meaning they won't notice on time.

As a result, the player plays games as Guest, realizing only after playing at least one game that the score was trashed due to that.

What should happen: a full screen notice should be shown, one that you'd have to confirm with clicking a button before being able to continue. Only if logout happened mid-game, the notice should then be postponed to be shown at the end of gameplay (or perhaps during intersection, where player could be asked to either ignore and continue playing or to quit/relogin).

#### 2. Unexpected logout
Only partially related to above, the logout itself was unexpected in the first place at the time, because I in fact did not log in on another device, unlike the message states. If I did a new login, I'd probably get an e-mail confirmation code like I did to re-log-in again...

### Screenshots or videos

![obraz](https://github.com/ppy/osu/assets/5182588/391a3eb2-190c-48e1-a483-44d4d446f06b)
^ should be a full screen notice instead of a notification

### Version

2024.221.0-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14423852/compressed-logs.zip)

Probably relevant fragment:
```
2024-02-27 17:31:34 [verbose]: Score submission token retrieved (189249286)
2024-02-27 17:31:39 [verbose]: 📺 OsuScreenStack#225(depth:5) suspended PlayerLoader#365 (waiting on SoloPlayer#148)
2024-02-27 17:31:39 [verbose]: 📺 OsuScreenStack#225(depth:6) entered SoloPlayer#148
2024-02-27 17:31:39 [verbose]: GameplayClockContainer seeking to -879
2024-02-27 17:31:43 [verbose]: Beginning score submission (token:189249286)...
2024-02-27 17:31:43 [verbose]: 📺 OsuScreenStack#225(depth:5) exit from SoloPlayer#148
2024-02-27 17:31:43 [verbose]: 📺 OsuScreenStack#225(depth:5) resume to PlayerLoader#365
2024-02-27 17:31:43 [verbose]: Score submission completed! (token:189249286 id:2418077730)
2024-02-27 17:31:43 [verbose]: ⚠️ You have been logged out on this device due to a login to your account on another device.
2024-02-27 17:31:44 [verbose]: 🖍️ Shader Texture2D/SpinnerGlow loaded from cache!
2024-02-27 17:31:44 [verbose]: 📺 OsuScreenStack#225(depth:5) suspended PlayerLoader#365 (waiting on SoloPlayer#239)
2024-02-27 17:31:44 [verbose]: 📺 OsuScreenStack#225(depth:6) entered SoloPlayer#239
2024-02-27 17:31:44 [verbose]: GameplayClockContainer seeking to -879
2024-02-27 17:34:27 [verbose]: No token, skipping score submission
2024-02-27 17:34:27 [verbose]: 📺 OsuScreenStack#225(depth:6) suspended SoloPlayer#239 (waiting on SoloResultsScreen#988)
2024-02-27 17:34:27 [verbose]: 📺 OsuScreenStack#225(depth:7) loading SoloResultsScreen#988
2024-02-27 17:34:27 [verbose]: 📺 OsuScreenStack#225(depth:7) entered SoloResultsScreen#988
2024-02-27 17:34:59 [verbose]: 📺 OsuScreenStack#225(depth:6) exit from SoloResultsScreen#988
2024-02-27 17:34:59 [verbose]: 📺 OsuScreenStack#225(depth:6) resume to SoloPlayer#239
2024-02-27 17:34:59 [verbose]: No token, skipping score submission
```
